talk/owt/sdk/p2p/p2ppeerconnectionchannel.cc

```diff
@@ -841,6 +854,11 @@ void P2PPeerConnectionChannel::Unpublish(
     std::function<void()> on_success,
     std::function<void(std::unique_ptr<Exception>)> on_failure) {
   rtc::scoped_refptr<webrtc::PeerConnectionInterface> temp_pc_ = GetPeerConnectionRef();
+  if (!temp_pc_) {
+    RTC_LOG(LS_INFO) << "Peer connection closed, returning.";
+    return;
+  }
```

Comment:
@davidtag We're also skipping a call to `DrainPendingSteams()` with this change. 
 In `DrainPendingStreams()` the real work only happens if the peer connection is still alive. So I think it's safe to skip, but perhaps we need to clear the instance variables `pending_publish_streams_` and `pending_unpublish_streams_`? Wdyt?

Reply:
the same is happening in Publish() fyi

Reply:
Yes, and I think it's okay to skip it there as well.

Reply:
I don't think we need to clear the vector. That should happen automatically when the `P2PPeerConnectionChannel` object is destroyed

Reply:
Yes, but there's a subtlety here. The vector holds shared pointers to `LocalStream` objects which means until we clear, the underlying `LocalStream` objects won't be destroyed. Previously this clearing would happen at these methods, but now we're deferring the clearing to `~P2PPeerConnectionChannel()` which might happen now or in the future. Is it safe to defer that? 

On a separate note, I noticed you're manually clearing the *_snapshot vectors in lines 1183-1184 and that's unnecessary. It'll get auto-cleaned up due to RAII.

Reply:
Actually that entire method is a bit messy and over-complicated in doing what it's trying to do by not taking advantage of modern cpp idioms. :-( But given our lack of test coverage, I'll refrain from refactoring it.